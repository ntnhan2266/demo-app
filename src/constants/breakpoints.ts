export const BREAKPOINTS = {
  small: 640,
  medium: 768,
  large: 1024,
  xlarge: 1280,
};
