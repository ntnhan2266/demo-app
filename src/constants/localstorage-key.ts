export const LOCALSTORAGE_KEY = {
  USERS: 'users'
}