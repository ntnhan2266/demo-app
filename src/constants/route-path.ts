export const ROUTE_PATH = {
  LOGIN: '/login',
  REGISTER: '/register',
  DASHBOARD: '/',
};
