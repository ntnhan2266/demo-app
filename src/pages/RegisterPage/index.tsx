import React from 'react';
import { Link } from 'react-router-dom';
import RegisterForm from '@/components/auth/RegisterForm';
import AppLogo from '@/components/common/AppLogo';
import { ROUTE_PATH } from '@/constants/route-path';
import withAuthentication from '@/hocs/withAuthentication';

const RegisterPage: React.FC = () => {
  return (
    <div data-testid='register-page' className='container mx-auto'>
      <div className='mb-4 max-w-md mx-auto mt-8 p-6 bg-white rounded-md shadow'>
        <div className='mb-6'>
          <AppLogo />
        </div>
        <RegisterForm />
      </div>
      <div>
        <p className='text-center'>
          Already have an account? &nbsp;
          <Link className='underline' to={ROUTE_PATH.LOGIN}>
            Login now.
          </Link>
        </p>
      </div>
    </div>
  );
};

export default withAuthentication(RegisterPage);
