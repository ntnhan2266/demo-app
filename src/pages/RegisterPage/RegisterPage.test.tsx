import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import RegisterPage from '@/pages/RegisterPage';
import { ROUTE_PATH } from '@/constants/route-path';
import renderer from 'react-test-renderer';

// Mock the withAuthentication HOC
jest.mock('@/hocs/withAuthentication', () => (Component: React.FC) => Component);

describe('RegisterPage', () => {
  test('should match the snapshot', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <RegisterPage />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders RegisterForm', () => {
    render(
      <MemoryRouter>
        <RegisterPage />
      </MemoryRouter>,
    );

    const registerFormElement = screen.getByTestId('register-form'); // Make sure to set the data-testid in RegisterForm component
    expect(registerFormElement).toBeInTheDocument();
  });

  test('renders link to Login page', () => {
    render(
      <MemoryRouter>
        <RegisterPage />
      </MemoryRouter>,
    );

    const loginLinkElement = screen.getByRole('link', { name: /login now/i });
    expect(loginLinkElement).toBeInTheDocument();
    expect(loginLinkElement).toHaveAttribute('href', ROUTE_PATH.LOGIN);
  });
});
