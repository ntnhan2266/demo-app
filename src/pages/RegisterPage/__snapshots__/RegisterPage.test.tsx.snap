// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterPage should match the snapshot 1`] = `
<div
  className="container mx-auto"
  data-testid="register-page"
>
  <div
    className="mb-4 max-w-md mx-auto mt-8 p-6 bg-white rounded-md shadow"
  >
    <div
      className="mb-6"
    >
      <a
        className="text-black font-bold font-comfortaa text-4xl"
        href="/"
        onClick={[Function]}
      >
        βeta
      </a>
    </div>
    <div
      data-testid="register-form"
    >
      <h2
        className="text-2xl font-bold mb-4 uppercase"
      >
        Create your account
      </h2>
      <form
        onSubmit={[Function]}
      >
        <div
          className="mb-4"
        >
          <label
            className="block w-full mb-1 text-xs font-medium text-gray-500 transition-all duration-200 ease-in-out "
            htmlFor="firstName"
          >
            First name
            <sup
              className="text-red-600"
            >
              *
            </sup>
          </label>
          <input
            className="border rounded-md px-3 py-2 focus:outline-none transition duration-300 w-full md:w-96 max-w-full w-full md:w-full border-gray-400"
            data-testid="first-name-input"
            disabled={false}
            name="firstName"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Enter your first name"
            type="text"
          />
        </div>
        <div
          className="mb-4"
        >
          <label
            className="block w-full mb-1 text-xs font-medium text-gray-500 transition-all duration-200 ease-in-out "
            htmlFor="lastName"
          >
            Last name
            <sup
              className="text-red-600"
            >
              *
            </sup>
          </label>
          <input
            className="border rounded-md px-3 py-2 focus:outline-none transition duration-300 w-full md:w-96 max-w-full w-full md:w-full border-gray-400"
            data-testid="last-name-input"
            disabled={false}
            name="lastName"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Enter your last name"
            type="text"
          />
        </div>
        <div
          className="mb-4"
        >
          <label
            className="block w-full mb-1 text-xs font-medium text-gray-500 transition-all duration-200 ease-in-out "
            htmlFor="emailOrPhone"
          >
            Email or phone
            <sup
              className="text-red-600"
            >
              *
            </sup>
          </label>
          <input
            className="border rounded-md px-3 py-2 focus:outline-none transition duration-300 w-full md:w-96 max-w-full w-full md:w-full border-gray-400"
            data-testid="email-or-phone-input"
            disabled={false}
            name="emailOrPhone"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Enter either your email or phone number"
            type="text"
          />
        </div>
        <div
          className="mb-4"
        >
          <label
            className="block w-full mb-1 text-xs font-medium text-gray-500 transition-all duration-200 ease-in-out "
            htmlFor="password"
          >
            Password
            <sup
              className="text-red-600"
            >
              *
            </sup>
          </label>
          <input
            className="border rounded-md px-3 py-2 focus:outline-none transition duration-300 w-full md:w-96 max-w-full w-full md:w-full border-gray-400"
            data-testid="password-input"
            disabled={false}
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Enter your password"
            type="password"
          />
        </div>
        <div
          className="mb-4"
        >
          <label
            className="block w-full mb-1 text-xs font-medium text-gray-500 transition-all duration-200 ease-in-out "
            htmlFor="confirmPassword"
          >
            Confirm password
            <sup
              className="text-red-600"
            >
              *
            </sup>
          </label>
          <input
            className="border rounded-md px-3 py-2 focus:outline-none transition duration-300 w-full md:w-96 max-w-full w-full md:w-full border-gray-400"
            data-testid="confirm-password-input"
            disabled={false}
            name="confirmPassword"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Confirm your password"
            type="password"
          />
        </div>
        <button
          className="rounded-md focus:outline-none w-full text-white text-base px-4 py-2 bg-indigo-500 hover:bg-indigo-600"
          data-testid="submit-button"
          disabled={false}
          type="submit"
        >
          Register
        </button>
      </form>
    </div>
  </div>
  <div>
    <p
      className="text-center"
    >
      Already have an account?  
      <a
        className="underline"
        href="/login"
        onClick={[Function]}
      >
        Login now.
      </a>
    </p>
  </div>
</div>
`;
